example/android/app/build.gradle

```diff
@@ -21,24 +23,33 @@ if (flutterVersionName == null) {
     flutterVersionName = '1.0'
 }
 
-apply plugin: 'com.android.application'
-apply plugin: 'kotlin-android'
-apply from: "$flutterRoot/packages/flutter_tools/gradle/flutter.gradle"
+def keystoreProperties = new Properties()
+def keystorePropertiesFile = rootProject.file('key.properties')
+if (keystorePropertiesFile.exists()) {
+    keystoreProperties.load(new FileInputStream(keystorePropertiesFile))
+}
 
 android {
     namespace "com.example.example"
+    compileSdk 34
+
+     compileOptions {
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
+    }
 
-    compileSdkVersion 33
+    kotlinOptions {
+        jvmTarget = '17'
+    }
 
     sourceSets {
         main.java.srcDirs += 'src/main/kotlin'
     }
 
     defaultConfig {
-        // TODO: Specify your own unique Application ID (https://developer.android.com/studio/build/application-id.html).
-        applicationId "com.example.example"
-        minSdkVersion 21
-        targetSdkVersion 33
+        applicationId  "com.example.example"
+        minSdk 24
```

Comment:
@vaishnavi-2301 Is there any reason why `minSdk` on Android needs to be set at `24`?

Unless you cite where it needs to be set that high, please set it to `21`, which is the bare minimum that Flutter now supports. Thanks.

Reply:
Okay will change it to the 21, i updated it according to your reference of wakelock_plus.